feat: Add WebSocket ping mechanism

psibi opened the pull request:
The WebSocket handler now includes a new background task to periodically send ping frames to connected clients. This addresses the issue of idle connections being closed by intermediaries like load balancers or proxies due to lack of activity.

I have been seeing logs like this in our internal service:

```
WebSocket protocol error: Connection reset without closing ha
Connection reset without closing handshake
```

The crate tungstenite automatically responds to the ping message according to what I read, so I haven't had to implement the pong part either in server or client - since we get that for free.

snoyberg commented on packages/kolme/src/pass_through.rs:
```diff
@@ -177,9 +179,30 @@ async fn ws_handler(ws: WebSocketUpgrade, State(state): State<PassThrough>) -> i
+                .await
+                .send(axum::extract::ws::Message::Ping(Vec::new().into()))
+                .await;
+            if res.is_err() {
```
```suggestion
            if let Err(e) = res {
                tracing::warn!("Client connection closed, stopping pings: {e}");
```

psibi commented:
@snoyberg Sorry, pushed a new change. Looks like I changed the pass through implementation instead of api-server. Updated it now along with addressing your comments. Requesting re-reviwe.
